Kernel/Arch/x86_64/ISABus/I8042Controller.cpp

```diff
@@ -358,6 +358,11 @@ ErrorOr<void> I8042Controller::do_reset_device(PortIndex port_index)
     // Wait until we get the self-test result
     auto self_test_result = TRY(do_wait_then_read_any_input(I8042Port::Buffer));
 
+    // Acknowledge means that reset is still in progress.
+    // Consume it and wait a bit longer
+    if (self_test_result == I8042Response::Acknowledge)
```

Comment:
On Discord, someone wondered if this should be a while instead of an if?

Reply:
Normally no AFAICT. Single reset command should issue a single ACK. Of course, real-life EC may have another opinion. Yet, I'm more concerned with a hang here than so buggy ECs